kore/src/Kore/Parser/Lexer.x

```diff
@@ -0,0 +1,314 @@
+  | TokenClaim
+  | TokenHookedSort
+  | TokenHookedSymbol
+  | TokenColonEqual
+  | TokenColon
+  | TokenLeftBrace
+  | TokenRightBrace
+  | TokenLeftBracket
+  | TokenRightBracket
+  | TokenLeftParen
+  | TokenRightParen
+  | TokenComma
+  | TokenTop
+  | TokenBottom
+  | TokenNot
+  | TokenAnd
+  | TokenOr
+  | TokenImplies
+  | TokenIff
+  | TokenExists
+  | TokenForall
+  | TokenMu
+  | TokenNu
+  | TokenCeil
+  | TokenFloor
+  | TokenEquals
+  | TokenIn
+  | TokenNext
+  | TokenRewrites
+  | TokenDomainValue
+  | TokenLeftAssoc
+  | TokenRightAssoc
+  | TokenIdent Text
+  | TokenSetIdent Text
+  | TokenString Text
+  | TokenEOF
+  deriving stock (Eq, Show)
+
+
+alexScanUser :: () -> AlexInput -> Int -> AlexReturn (AlexInput -> Int -> Alex Token)
+
+data AlexInput = AlexInput { alexPosn :: {-# UNPACK #-} !AlexPosn,
+                             alexChar :: {-# UNPACK #-} !Char,
+                             alexStr :: {-# UNPACK #-} !ByteString.ByteString,
+                             alexBytePos :: {-# UNPACK #-} !Int}
+
+data AlexPosn = AlexPn !FilePath !Int !Int !Int
+        deriving (Eq,Show)
+
+data AlexState = AlexState {
+        alex_pos :: !AlexPosn,  -- position at current input location
+        alex_bpos:: !Int,     -- bytes consumed so far
+        alex_inp :: ByteString.ByteString,      -- the current input
+        alex_chr :: !Char,      -- the character before the input
+        alex_scd :: !Int        -- the current startcode
+    }
+
+newtype Alex a = Alex { unAlex :: AlexState -> Either String (AlexState, a) }
+
+instance Functor Alex where
+  fmap f a = Alex $ \s -> case unAlex a s of
+                            Left msg -> Left msg
+                            Right (s', a') -> Right (s', f a')
+
+instance Applicative Alex where
+  pure a   = Alex $ \s -> Right (s, a)
+  fa <*> a = Alex $ \s -> case unAlex fa s of
+                            Left msg -> Left msg
+                            Right (s', f) -> case unAlex a s' of
+                                               Left msg -> Left msg
+                                               Right (s'', b) -> Right (s'', f b)
+
+instance Monad Alex where
+  m >>= k  = Alex $ \s -> case unAlex m s of
+                                Left msg -> Left msg
+                                Right (s',a) -> unAlex (k a) s'
+  return = App.pure
+
+alexGetByte (AlexInput {alexPosn=p,alexStr=cs,alexBytePos=n}) =
+    case ByteString.uncons cs of
+        Nothing -> Nothing
+        Just (c, rest) -> 
+          let b = ByteString.w2c c in
+            Just (c, AlexInput {
+                alexPosn = alexMove p b,
+                alexChar = b,
+                alexStr =  rest,
+                alexBytePos = n+1})
+
+alexMove :: AlexPosn -> Char -> AlexPosn
+alexMove (AlexPn fp a l c) '\t' =
+    AlexPn fp (a+1) l (c+alex_tab_size-((c-1) `mod` alex_tab_size))
+alexMove (AlexPn fp a l _) '\n' = AlexPn fp (a+1) (l+1) 1
+alexMove (AlexPn fp a l c) _    = AlexPn fp (a+1) l (c+1)
```

Comment:
All these 1-letter or 2-letter variable names really make things hard to read. We prefer longer, more suggestive variable names in order to make code as readable as possible.

Reply:
Same deal here.